opm/input/eclipse/Deck/FileDeck.cpp

```diff
@@ -202,8 +213,12 @@ const DeckKeyword& FileDeck::operator[](const Index& index) const
 
 void FileDeck::Block::load(const Deck& deck, std::size_t deck_index)
 {
-    const auto current_file = deck[deck_index].location().filename;
-
+    const auto& loc = deck[deck_index].location();
+    const auto& current_file = loc.filename;
+    this->is_binary = loc.is_binary();
+    if (is_binary) {
+        fname = current_file;
+    }
```

Comment:
I don't understand what this statement/block is supposed to do.  Would you mind elaborating?

As far as I can tell, the `Block` (i..e,  `*this`) is constructed in the calling scope (the `FileDeck::FileDeck(const Deck&)` constructor) with an argument that's exactly the same as this `current_file`.  Why do you need to reset `fname` here?

Reply:
You're right, this is completely unnecessary. Removing...